docs: dca first draft (#343)
chainflip-io/chainflip-docs: pages/swapping/swapping-basics.mdx

Minimum accepted prices are enforced on the AMM level and do not take deposit fees and broadcast fees into account.
If a deposit is refunded, the Chainflip protocol deducts a deposit fee and a broadcast fee from the refund amount to pay for the transactions sent by the protocol. If the deposit address has a broker fee, it will be deducted from the refund amount.

> ## Trading execution time for better price: DCA

Reviewer: ```suggestion
## Dollar Cost Average (DCA): Improving price
```

Any swap can be broken down into smaller "chunks" which are executed over time.
This results in longer execution time, but reduces certain risks for LPs which should translate into better pricing for the end user.
The chunks are executed according to DCA parameters, which can be set upon opening a [deposit channel](../../concepts/swaps-amm/deposit-channels-and-brokers),
and include the number of chunks and the interval between their execution.

If [slippage protection](minimum-accepted-price-slippage-protection) is enabled, the minimum price requirement is applied to each
individual chunk.
If a chunk is executed, but does not meet the minimum price requirement, it is retried a few blocks later, pushing back the
execution of any remaining chunks.

> If the same chunks reaches its retry limit, it will be refunded along with any remaining chunks in a single refund transaction.

Reviewer: ```suggestion
If the same chunk reaches its retry limit, it will be refunded along with any remaining chunks in a single transaction to the user-specified refund address.
```

> In that case, any amount already successfully executed in prior chunks is still sent to the user specified desitnation address.

Reviewer: ```suggestion
In that case, any amount successfully swapped in prior chunks is still sent to the user-specified destination address.
```

Minimum accepted prices are enforced on the AMM level and do not take deposit fees and broadcast fees into account.
If a deposit is refunded, the Chainflip protocol deducts a deposit fee and a broadcast fee from the refund amount to pay for the transactions sent by the protocol. If the deposit address has a broker fee, it will be deducted from the refund amount.

> ## Dollar Cost Average (DCA): Improving price

Reviewer 2: should we put this in its own section? like how we have Introduction to boost? I think its a bit difficult to find right now?

Author: I don't have an opinion here, probably a question for @fernandezdavid.

In that case, any amount successfully swapped in prior chunks is still sent to the user-specified destination address.

### Example 1: Successful DCA swap

1. A user initiates a swap by depositing **5 BTC** to get **USDC** with a minimum accepted price of **50,000 USDC/BTC**.
The channel's DCA parameters have been set to split the swap into **5 chunks** with an interval of **2 blocks** between them.
2. The deposit is witnessed at state chain block **100**.
The first chunk of **1 BTC** is scheduled for execution at block **102** (small initial delay allows LPs update their orders for the first chunk).
3. At block **102** the chunk is executed with the output of **50,100 USDC**, which is within the price limit.
According to DCA parameters, the second chunk is scheduled for execution **2 blocks** later, at block **104**.
4. At block **104** the second chunk is executed at a price of **49,900 USDC**, which is below the limit, and the swap is reverted.
According to [slippage protection](minimum-accepted-price-slippage-protection) parameters the chunk is to be retried a few times before the swap is refunded,
with the first atttempt scheduled for block **109**.

> 5. At block **109** the second chunk is rertied, and this time gives the user the price of **50,070 USDC**, which is within the limit.

Reviewer 3: `retried` typo
